jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/SessionHandler.java

```diff
@@ -717,41 +717,47 @@ public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingMode
     @Override
     public Request.Processor handle(Request request) throws Exception
     {
-        ServletContextRequest servletContextRequest = Request.as(request, ServletContextRequest.class);
-        ServletContextRequest.ServletApiRequest servletApiRequest =
-            (servletContextRequest == null ? null : servletContextRequest.getServletApiRequest());
-        if (servletApiRequest == null)
-            throw new IllegalStateException("Request is not a valid ServletContextRequest");
-
         Request.Processor processor = getHandler().handle(request);
         if (processor == null)
             return null;
 
-        addSessionStreamWrapper(request);
+        return new SessionProcessor(processor);
```

Comment:
No, because we don't wrap the request, so we have nowhere to hold the next processor.    This is a very light/cheap wrapper, so let's see what the performance testing says.

Reply:
Hmm it can be done with a setter/getter on ServletContextRequest.  A bit ugly, but have a look and tell me what you think